lucille-core/src/main/java/com/kmwllc/lucille/message/KafkaUtils.java

```diff
@@ -39,20 +41,31 @@ public class KafkaUtils {
   public static final Duration POLL_INTERVAL = Duration.ofMillis(2000);
   private static final Logger log = LoggerFactory.getLogger(KafkaUtils.class);
 
-  private static Properties loadExternalProps(String filename) {
-    try (Reader propertiesReader = FileUtils.getReader(filename, StandardCharsets.UTF_8.name())) {
-      Properties consumerProps = new Properties();
-      consumerProps.load(propertiesReader);
-      return consumerProps;
-    } catch (IOException e) {
-      throw new IllegalStateException(String.format("Cannot load kafka property file %s.", filename));
+  private static Properties loadExternalProps(String filename, Map<String, Object> cloudOptions) {
+    FileContentFetcher fetcher = new FileContentFetcher(cloudOptions);
```

Comment:
for use cases like this, we should investigate making a FileContentFetcher.getReader method that take cloudOptions as an arg and does the startup and shutdown logic internally...